File: HelloMetal/BasicBuffers/BasicBuffers/ViewController.m
//
//  ViewController.m
//  BasicBuffers
//
//  Created by meitu on 2018/7/26.
//  Copyright © 2018年 meitu. All rights reserved.
//

#import "ViewController.h"
#import "HMXRender.h"

@interface ViewController ()

@property (strong, nonatomic) MTKView *mtkView;
@property (strong, nonatomic) HMXRender *render;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.mtkView = (MTKView *)self.view;
    
    // 获取设备的默认GPU对象
    self.mtkView.device = MTLCreateSystemDefaultDevice();
    
    if (!self.mtkView.device) {
        NSLog(@"Metal is not supported on this device!");
        return;
    }
    
    self.render = [[HMXRender alloc] initWithMetalKitView:self.mtkView];
    
    if (!self.render) {
        NSLog(@"Renderer failed initializetion!");
        return;
    }
    
    [self.render mtkView:self.mtkView drawableSizeWillChange:self.mtkView.drawableSize];
    
    self.mtkView.delegate = self.render;
    
    self.mtkView.preferredFramesPerSecond = 60;
}

@end

File: HelloMetal/Metal01_Triangle/Metal01_Triangle/ViewController.swift
//
//  ViewController.swift
//  Metal01_Triangle
//
//  Created by meitu on 2018/7/31.
//  Copyright © 2018年 meitu. All rights reserved.
//

import UIKit
import MetalKit

class ViewController: UIViewController {
    
//    var mtkView:MTKView! = nil
//    var renderer:HMXRenderer! = nil
    
    public var renderer: Renderer? = nil
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let mtkView = self.view as! MTKView
        
        renderer = Renderer.init(mtkView: mtkView)
        if renderer == nil {
            print("Renderer failed initializetion!")
            return
        }
        
//        mtkView = self.view as! MTKView
//
//        mtkView.preferredFramesPerSecond = 60
//
//        mtkView.device = MTLCreateSystemDefaultDevice()
//        if (mtkView.device == nil) {
//            print("Metal is not supported on this device!")
//            return
//        }
//
//        renderer = HMXRenderer.init(mtkView: mtkView)
//        if renderer == nil {
//            print("Renderer failed initializetion!")
//            return
//        }
    }
    
}

